### PR TITLE
Simple utility to turn parameters into dimensions

### PR DESCRIPTION
**Pull request by jlstevens**
This PR aims to address #1190. 

Other than finding a better name for the utility, everything is straightforward (see the tests for an example of how to use it). Ready for review/merge once the tests are green.

### CODE REVIEW

File: holoviews/util.py [hunk 1]
```diff
@@ -127,3 +127,42 @@ def _make_dynamic(self, hmap, dynamic_fn, streams):
...
+    def __call__(self, parameterized, **kwargs):
+
+        types = {param.String:str, param.Integer:int}
+        p = param.ParamOverrides(self, kwargs)
+        dim_list = []
+        params = parameterized.params()
+        for name, param_obj in params.items():
+            if name in p.exclude:  continue
+            elif param_obj.precedence  and param_obj.precedence < 0: continue
+            dim_kws = {}
```
Comment (jbednar) [inline holoviews/util.py]: Can the precedence threshold be a parameter (defaulting to 0)?

**Comment by jbednar** [comment]
Looks good to me, thanks!!

File: holoviews/util.py (hunk 1 above)
Comment (jlstevens) [inline holoviews/util.py]: Sure. Addressed in f125e19d.

**Comment by philippjfr** [comment]
I question a little how useful this is in practice, it seems to cover a fairly limited number of parameters. At minimum ObjectSelector should also be handled, although I'm not sure yet how well arbitrary objects will work as Dimension.values, which is something we should probably handle in some way.

File: holoviews/util.py [hunk 2]
```diff
@@ -127,3 +127,47 @@ def _make_dynamic(self, hmap, dynamic_fn, streams):
...
+            if param_obj.doc and p.label:
+                dim_kws['label'] = param_obj.doc
+            if isinstance(param_obj, param.Number):
+                if param_obj.bounds != None:
+                    dim_kws['range'] = param_obj.bounds
+                if param_obj.softbounds != None:
+                    dim_kws['soft_range'] = param_obj.softbounds
+            elif isinstance(param_obj, param.List):
+                dim_kws['values'] = param_obj.default
```
Comment (philippjfr) [inline holoviews/util.py]: This doesn't make sense to me. A ``param.List`` expects the whole list as input but this treats it basically as an ObjectSelector.

**Comment by jlstevens** [comment]
I don't have any strong feelings about this utility - I see this PR as a skeleton for what (I hope!) will remain a fairly simple utility. Feel free to think about how you might want to use this in practice and push whatever improvements you think make sense.

**Comment by philippjfr** [comment]
In its current form it doesn't seem to handle some of the basic cases, in particular ObjectSelector and in some cases e.g. ``param.List`` arguably doesn't make sense. The main question I have is how we expect this to be used. Dimensions will never be able to cover all the cases that parameters can handle.

So in my opinion this should be skipping cases that won't work as Dimensions such as ListSelector/List/Tuple/NumericTuple etc. (they will all break when used as a HoloMap/DynamicMap because they are not valid keys). Because of this this utility will only ever cover a narrow range of cases. If you and @jbednar still think that's useful I'd be happy to merge it, but not without doing that validation.

**Comment by jlstevens** [comment]
> I question a little how useful this is in practice...

I also wonder about that but I don't object to having a *small* utility that is useful for simple cases. I don't see how to promote all the various parameter types to dimension but I do think many of the most common parameter case can be covered.

**Comment by jlstevens** [comment]
I posted my comment above at the same time as you posted yours. I think we essentially agree and it is up to @jbednar to decide...

**Comment by philippjfr** [comment]
I'm going to close this PR, we can revisit this later if we think there's a general approach that's not fundamentally limited.

**Comment by jlstevens** [comment]
Fine by me!

**Pull request closed without merging**